File: client/src/app/models/order-list-params.ts
import { PaginationParams } from "./pagination-params";

export enum OrderListType
{
    All = 0,
    SellOnly = 1,
    BuyOnly = 2
}

export class OrderListParams extends PaginationParams{
    typeFilter = OrderListType.All;
    showIncompleteOnly = false;
}

File: API/Pagination/OrderListParams.cs
using API.Entities;

namespace API.Pagination;

public enum OrderListType
{
    All = 0,
    SellOnly = 1,
    BuyOnly = 2
}

public class OrderListParams : PaginationParams
{
    public OrderListType TypeFilter { get; set; } = OrderListType.All;
    public bool ShowIncompleteOnly { get; set; } = false;
}